### PR TITLE
[Fixes AB#696] Set list item to look clickable (withdrawals/deposits/activations)

### PR DESCRIPTION
**Pull request by mdrusko1**

### CODE REVIEW

File: src/app/admin/manage-deposits/manage-deposits.component.html [hunk 1]
```diff
@@ -33,14 +33,16 @@
...
                     <span class="float-left explainer-mini">
                         {{ deposit.deposit.reference }} ({{ deposit.user.email }})
                     </span>
-                    <span (click)="declineDeposit(deposit.deposit.id).subscribe()" class="float-right cursor-pointer">
+                    <a [routerLink]="">
+                    <span (click)="declineDeposit(deposit.deposit.id).subscribe()" class="float-right">
                         <i class="fas fa-trash-alt"></i>
                     </span>
-                    <a class="float-right mr-4 cursor-pointer" href="mailto:{{ deposit.user.email }}">
+                    </a>
```
Comment (mightymatth) [inline src/app/admin/manage-deposits/manage-deposits.component.html:44]: Click event should be always added to an anchor tag. This leads to breaking the semantics. Is there a visual reason for that?

File: src/app/admin/manage-withdrawals/manage-withdrawals.component.html [hunk 2]
```diff
@@ -6,22 +6,25 @@ <h2 class="app-page-subtitle">
...
+            <ul *ngIf="userWithdrawals.length > 0" class="list-group">
+                <li *ngFor="let userWithdrawal of userWithdrawals"
+                    [routerLink]="['/dash', 'manage_withdrawals', userWithdrawal.withdraw.id] | coopPath"
+                    class="list-group-item link-remove-style gray-text explainer-mini">
+                    <span class="float-left">
+                     {{ userWithdrawal.user.first_name }} {{ userWithdrawal.user.last_name }}
+                        ({{ userWithdrawal.user.email }})
                 </span>
-                <span class="float-right">
+                    <span class="float-right">
```
Comment (mightymatth) [inline src/app/admin/manage-withdrawals/manage-withdrawals.component.html:18]: this is wrongly aligned.

File: src/app/admin/platform-bank-account/platform-bank-account.component.html [hunk 3]
```diff
@@ -29,14 +29,16 @@ <h1 class="app-page-title">
...
+                            <div class="mt-1 link-remove-style">
                                 <div class="explainer-mini text-uppercase mb-2">
                                     {{ 'admin.bank_account.item.actions' | translate }}
                                 </div>
+                                <a [routerLink]="">
                                 <span (click)="deleteBankAccountClicked(paymentOption.id)"
                                       class="float-right align-middle cursor-pointer">
 								<i class="fas fa-trash-alt"></i>
 							</span>
+                                </a>
```
Comment (mightymatth) [inline src/app/admin/platform-bank-account/platform-bank-account.component.html:41]: again, click should be on the anchor tag.

File: src/app/admin/wallet-activation/group-activation/group-activation.component.html [hunk 4]
```diff
@@ -4,13 +4,15 @@ <h5 class="explainer-mini">
...
+            <li *ngFor="let group of groups" class="list-group-item explainer-mini link-remove-style">
                 {{ group.organization.name }}
+                <a [routerLink]="">
                 <span (click)="activateGroup(group.wallet.uuid).subscribe()"
-                      class="float-right cursor-pointer hover-fg">
+                      class="float-right hover-fg">
                     <i class="fas fa-check-circle mr-2"></i>
                     {{ 'admin.activations.groups.activate' | translate }}
                 </span>
+                </a>
```
Comment (mightymatth) [inline src/app/admin/wallet-activation/group-activation/group-activation.component.html:15]: click on the anchor tag.

File: src/app/admin/wallet-activation/project-activation/project-activation.component.html [hunk 5]
```diff
@@ -4,13 +4,15 @@ <h5 class="explainer-mini">
...
+            <li *ngFor="let project of projects" class="list-group-item explainer-mini link-remove-style">
                 {{ project.project.name }}
+                <a [routerLink]="">
                 <span (click)="activateProject(project.wallet.uuid).subscribe()"
-                      class="float-right cursor-pointer hover-fg">
+                      class="float-right hover-fg">
                     <i class="fas fa-check-circle mr-2"></i>
                     {{ 'admin.activations.projects.activate' | translate }}
                 </span>
+                </a>
```
Comment (mightymatth) [inline src/app/admin/wallet-activation/project-activation/project-activation.component.html:15]: click on the anchor tag.

File: src/app/admin/wallet-activation/user-activation/user-activation.component.html [hunk 6]
```diff
@@ -4,13 +4,15 @@ <h5 class="explainer-mini">
...
+            <li *ngFor="let user of users" class="list-group-item explainer-mini link-remove-style">
                 {{ user.user.first_name }} {{ user.user.last_name }} (<i>{{ user.user.email }}</i>)
+                <a [routerLink]="">
                 <span (click)="activateUserClicked(user.wallet.uuid).subscribe()"
-                      class="float-right cursor-pointer hover-fg">
+                      class="float-right hover-fg">
                     <i class="fas fa-check-circle mr-2"></i>
                     {{ 'admin.activations.users.activate' | translate }}
                 </span>
+                </a>
```
Comment (mightymatth) [inline src/app/admin/wallet-activation/user-activation/user-activation.component.html:15]: click on the anchor tag

File: src/app/organizations/organization-details/manage-projects/manage-single-project/manage-single-project.component.html [hunk 7]
```diff
@@ -214,10 +214,12 @@ <h4>{{ states.project.name }}</h4>
                                                                                         {{ file.name }}
                                                                                     </div>
                                                                                     <div class="docs__icons">
+                                                                                        <a [routerLink]="">
                                                                                         <span (click)="deleteFile(states.project, file.id).subscribe()"
                                                                                               class="mr-3 cursor-pointer gray-text">
                                                                                         <i class="fas fa-trash"></i>
                                                                                         </span>
+                                                                                        </a>
```
Comment (mightymatth) [inline src/app/organizations/organization-details/manage-projects/manage-single-project/manage-single-project.component.html:222]: click on the anchor tag.

File: src/styles/_common.scss [hunk 8]
```diff
@@ -171,6 +173,27 @@
...
+.list-remove-style {
+  ul {
+    list-style-type: none;
+  }
+
+  li {
+    cursor: pointer;
+  }
+}
+
```
Comment (mightymatth) [inline src/styles/_common.scss:196]: this class does not only remove the style (how it says) but also adds a cursor pointer. why is that pointer necessary of we said that we'll use anchor tags for anything that is clickable?

**Review by mightymatth — CHANGES_REQUESTED** [review]

File: src/styles/_common.scss (hunk 8 above)
Comment (mdrusko1) [reply src/styles/_common.scss:196]: In this case, specifically, we need to have the cursor pointer over the whole row (li element) and the [routerLink]
inside which onClick opens the next component.
So when I try to wrap this element with an anchor tag or replace < li > with < a > the IDE gives me a warning 'Element a is not allowed here', so I created _list-remove-style_  class selector (maybe I should give another name as you mentioned) to be able to get appropriate styling. Also, we can use this selector in other places in a project where needed. 
But, If there is a better solution for that, I would definitely like to hear it. ;-) 

image

File: src/app/admin/manage-deposits/manage-deposits.component.html (hunk 1 above)
Comment (mdrusko1) [reply src/app/admin/manage-deposits/manage-deposits.component.html:44]: OK thnx, will fix it.

**Review by mightymatth — COMMENTED** [review]
I've made some changes to avoid plenty of comments.

1. In [this](https://github.com/AMPnet/ampnet-crowdfunding-frontend/pull/171/commits/6c2cba79c40856878c7aeaec6ba0fa5d7d824c24) commit, I've aligned wrongly indented spans. Intellij is kinda stupid about them :P 

2. In [this](https://github.com/AMPnet/ampnet-crowdfunding-frontend/pull/171/commits/4c779fe39953a349aaf13d4e46f7ae5ee678caec) commit, I fixed list items in admin withdrawals screen. (I'll comment it in our discussion).

File: src/styles/_common.scss (hunk 8 above)
Comment (mightymatth) [reply src/styles/_common.scss]: Suggestion => Bootstrap docs: https://getbootstrap.com/docs/4.0/components/list-group/#links-and-buttons

No need for using `ul` and `li`.

**Review by mightymatth — APPROVED** [review]
I made more changes. Most of them are trimming white spaces in class names and removing the obsolete `explainer-mini` class from the entire app.

What I did importantly, there will be some comment.

File: src/styles/_overrides.scss [hunk 9]
```diff
@@ -1,3 +1,25 @@
...
+  &:hover {
+    color: darken($primary-color, 7.5%);
+  }
+
+  &.nostyle {
+    text-decoration: inherit !important;
+    color: inherit !important;
+    cursor: auto !important;
+  }
+}
```
Comment (mightymatth) [inline src/styles/_overrides.scss:20]: I've made a global change on every link item to match the primary color of the app. It looks really nice and it's convenient that a link looks different than normal text. If you want to override it and match the original style, use `nostyle` class on a particular link.

File: src/app/admin/wallet-activation/project-activation/project-activation.component.html [hunk 10]
```diff
@@ -1,10 +1,10 @@
 <div *ngIf="projects !== undefined">
-    <h5 class="explainer-mini">
+    <h5>
         {{ 'admin.activations.projects.title' | translate }}
     </h5>
     <div *ngIf="projects.length > 0" class="mt-3">
         <ul class="list-group">
-            <li *ngFor="let project of projects" class="list-group-item explainer-mini link-remove-style">
```
Comment (mightymatth) [inline src/app/admin/wallet-activation/project-activation/project-activation.component.html]: There is no reason to add `link-remove-style` class on the list item if you apply that on the anchor link. (that's why I made `nostyle` on a specific link.

File: src/styles/_overrides.scss (hunk 9 above)
Comment (mightymatth) [reply src/styles/_overrides.scss:20]: @Skafec just to mention you here to see this `nostyle` attribute to reset a link style.

**Comment by mdrusko1** [comment]
At first, the task itself seemed so simple. 
It was just needed to set a cursor pointer. In the end, many changes came out, and I learned some new things today. ;) Thnx.

File: src/styles/_overrides.scss (hunk 9 above)
Comment (mdrusko1) [reply src/styles/_overrides.scss:20]: Great idea 👍

**Pull request merged**